hudi-common/src/test/java/org/apache/hudi/common/table/TestTimelineUtils.java

```diff
@@ -181,10 +187,113 @@ public void testRestoreInstants() throws Exception {
+      instants.add(instant);
+    }
+
+    HoodieRollbackMetadata rollbackMetadata = TimelineMetadataUtils.convertRollbackMetadata("001", Option.of(1234L),
+        instants, rollbackStats);
+
+    Option<byte[]> ser = TimelineMetadataUtils.serializeRollbackMetadata(rollbackMetadata);
+    if (ser.isPresent()) {
+      HoodieRollbackMetadata rollbackMetadata1 = TimelineMetadataUtils.deserializeHoodieRollbackMetadata(ser.get());
+      System.out.println("deser successfully ");
+      for (Entry<String, HoodieRollbackPartitionMetadata> rollbackStat : rollbackMetadata1.getPartitionMetadata().entrySet()) {
+        System.out.println("Deser value fro " + rollbackStat.getKey());
+        HoodieRollbackPartitionMetadata stat = rollbackStat.getValue();
+        System.out.println(
+            "1111 Rollback stat for partition " + stat.getPartitionPath() + " success files " + (stat.getSuccessDeleteFiles() != null ? Arrays
+                .toString(stat.getSuccessDeleteFiles().toArray(new String[0]))
+                : "null")
+                + ", delete files :: " + (stat.getSuccessDeleteFiles() != null ? Arrays.toString(stat.getFailedDeleteFiles().toArray(new String[0])) : "null")
+                + ", rollback log files " + stat
+                .getRollbackLogFiles().entrySet()
+                + ", probable/written log files : " + stat.getWrittenLogFiles().entrySet());
+      }
+    }
+
+    System.out.println("---------------------------------------------------------------------------");
+    System.out.println("---------------------------------------------------------------------------");
+    System.out.println("---------------------------------------------------------------------------");
+
+    Map<String, List<HoodieRollbackMetadata>> instantToRollbackMetadata = new HashMap<>();
+    instantToRollbackMetadata.put("abc", Collections.singletonList(rollbackMetadata));
+
+    HoodieRestoreMetadata restoreMetadata = TimelineMetadataUtils.convertRestoreMetadata("002", 10L,
+        instants, instantToRollbackMetadata);
+    ser = TimelineMetadataUtils.serializeRestoreMetadata(restoreMetadata);
+    if (ser.isPresent()) {
+      HoodieRestoreMetadata restoreMetadata1 = TimelineMetadataUtils.deserializeHoodieRestoreMetadata(ser.get());
+      System.out.println("deser successfully ");
+
+      for (Entry<String, List<HoodieRollbackMetadata>> entry : restoreMetadata1.getHoodieRestoreMetadata().entrySet()) {
+        List<HoodieRollbackMetadata> rollbackMetadatas = entry.getValue();
+        for (HoodieRollbackMetadata rollbackMetadata2 : rollbackMetadatas) {
+          for (Entry<String, HoodieRollbackPartitionMetadata> rollbackStat : rollbackMetadata2.getPartitionMetadata().entrySet()) {
+            System.out.println("Deser value fro " + rollbackStat.getKey());
+            HoodieRollbackPartitionMetadata stat = rollbackStat.getValue();
+            System.out.println(
```

Comment:
![image](https://user-images.githubusercontent.com/1179324/104044081-a5480380-5191-11eb-90ec-9113721cd1bc.png)
 Can you try setting a breakpoint here and see if these fields actually get written into the outputstream. 

My guess is they are. but the schema used for reading is different. Make sure you recompile the entire project from command line once. `mvn clean install ....` before running the test via IDE